pkg/api/types.go

```diff
@@ -3577,8 +3580,78 @@ type Event struct {
 	// Type of this event (Normal, Warning), new types could be added in the future.
 	// +optional
 	Type string
+
+	// Time when this Event was first observed.
+	// +optional
+	EventTime metav1.MicroTime
+
+	// Data about the Event series this event represents or nil if it's a singleton Event.
+	// +optional
+	Series *EventSeries
+
+	// Component that generated the Event.
+	// +optional
+	Origin EventSource
```

Comment:
Why aren't you reusing `Source` field above, which I guess is meant for the same purpose?